core/src/lib.rs

```diff
@@ -139,6 +138,12 @@ pub enum AttributeBitFlags {
     UniqueValue   = 1 << 3,
 }
 
+#[derive(Clone,Debug,Eq,Hash,Ord,PartialOrd,PartialEq)]
+pub enum Unique {
```

Comment:
Thanks for correcting me on this; I should have known better.  I've pushed it into an `attribute` submodule.